src/client.rs

```diff
@@ -256,24 +253,54 @@ impl<'a> RequestBuilder<'a> {
 
 /// A Response to a submitted `Request`.
 #[derive(Debug)]
-pub struct Response {
-    inner: ::hyper::client::Response,
+pub enum Response {
```

Comment:
Let's keep the internals private; enum variants are all public. So instead, something like:

```rust
pub struct Response {
    inner: Decoder,
}

enum Decoder {
    Plaintext(::hyper::client::Response),
    Gzip(..),
}
```